CC/Sounder/Radio.cc

```diff
@@ -6,33 +6,61 @@
+        dev->setFrequency(SOAPY_SDR_RX, ch, "RF", _cfg->radioRfFreq);
+        dev->setFrequency(SOAPY_SDR_RX, ch, "BB", 0);
+        dev->setFrequency(SOAPY_SDR_TX, ch, "RF", _cfg->radioRfFreq);
+        dev->setFrequency(SOAPY_SDR_TX, ch, "BB", 0);
+    }
 
-    dev->setFrequency(SOAPY_SDR_RX, ch, "RF", _cfg->radioRfFreq);
-    dev->setFrequency(SOAPY_SDR_RX, ch, "BB", _cfg->nco);
-    dev->setFrequency(SOAPY_SDR_TX, ch, "RF", _cfg->radioRfFreq);
-    dev->setFrequency(SOAPY_SDR_TX, ch, "BB", _cfg->nco);
+    std::cout << "Tx RF freq: " << dev->getFrequency(SOAPY_SDR_TX, ch, "RF")
+              << ", Rx RF freq: " << dev->getFrequency(SOAPY_SDR_RX, ch, "RF") << std::endl;
+    std::cout << "Tx BB freq: " << dev->getFrequency(SOAPY_SDR_TX, ch, "BB")
+              << ", Rx BB freq: " << dev->getFrequency(SOAPY_SDR_RX, ch, "BB") << std::endl;
 
     // Unified gains for both lime and frontend
     if (_cfg->single_gain) {
         dev->setGain(SOAPY_SDR_RX, ch, rxgain); // w/CBRS 3.6GHz [0:105], 2.5GHZ [0:108]
         dev->setGain(SOAPY_SDR_TX, ch, txgain); // w/CBRS 3.6GHz [0:105], 2.5GHZ [0:105]
+        std::cout << "Tx gain: " << dev->getGain(SOAPY_SDR_TX, ch)
+                  << ", Rx gain: " << dev->getGain(SOAPY_SDR_RX, ch) << std::endl;
     } else {
-	dev->setGain(SOAPY_SDR_RX, ch, "LNA", std::min(30.0, rxgain)); // w/CBRS 3.6GHz [0:105], 2.5GHZ [0:108]
-        dev->setGain(SOAPY_SDR_RX, ch, "TIA", 0);
-        dev->setGain(SOAPY_SDR_RX, ch, "PGA", 0);
-        dev->setGain(SOAPY_SDR_RX, ch, "LNA2", 17); // w/CBRS 3.6GHz [0:105], 2.5GHZ [0:108]
-        dev->setGain(SOAPY_SDR_RX, ch, "ATTN", _cfg->radioRfFreq < 3e9 ? -12 : 0);
-        dev->setGain(SOAPY_SDR_TX, ch, "PAD", std::min(42.0, txgain)); // w/CBRS 3.6GHz [0:105], 2.5GHZ [0:105]
-        dev->setGain(SOAPY_SDR_TX, ch, "IAMP", 0);
-        dev->setGain(SOAPY_SDR_TX, ch, "PA2", 0);
-        dev->setGain(SOAPY_SDR_TX, ch, "ATTN", -6);
+        if (!kUseUHD) {
+            dev->setGain(SOAPY_SDR_RX, ch, "LNA", std::min(30.0, rxgain)); // w/CBRS 3.6GHz [0:105], 2.5GHZ [0:108]
+            dev->setGain(SOAPY_SDR_RX, ch, "TIA", 0);
+            dev->setGain(SOAPY_SDR_RX, ch, "PGA", 0);
+            dev->setGain(SOAPY_SDR_RX, ch, "LNA2", 17); // w/CBRS 3.6GHz [0:105], 2.5GHZ [0:108]
+            dev->setGain(SOAPY_SDR_RX, ch, "ATTN", _cfg->radioRfFreq < 3e9 ? -12 : 0);
+            dev->setGain(SOAPY_SDR_TX, ch, "PAD", std::min(42.0, txgain)); // w/CBRS 3.6GHz [0:105], 2.5GHZ [0:105]
+            dev->setGain(SOAPY_SDR_TX, ch, "IAMP", 0);
+            dev->setGain(SOAPY_SDR_TX, ch, "PA2", 0);
+            dev->setGain(SOAPY_SDR_TX, ch, "ATTN", -6);
+        } else {
+            dev->setGain(SOAPY_SDR_RX, ch, rxgain);
+            dev->setGain(SOAPY_SDR_TX, ch, txgain);
+            std::cout << "Tx gain: " << dev->getGain(SOAPY_SDR_TX, ch) << std::endl;
+            std::cout << "Rx gain: " << dev->getGain(SOAPY_SDR_RX, ch) << std::endl;
```

Comment:
We can move this out of this if as well.

Reply:
Done